biz.aQute.bndlib/src/aQute/bnd/service/Tagged.java

```diff
@@ -0,0 +1,60 @@
+package aQute.bnd.service;
+
+import static aQute.bnd.service.Tagged.RepoTags.resolve;
+import static java.util.stream.Collectors.toCollection;
+
+import java.util.Arrays;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+/**
+ * Allows to add tags to implementing classes. Originally intended for tagging
+ * repositories.
+ */
+public interface Tagged {
+
+
+	enum RepoTags {
```

Comment:
Why not move this to the RepositoryPlugin? The repositories can know about tags but I am a bit uncomfortable with tags knowing about repositories. Seems to break through the layering. And I always like to consider what happens when we get 50 subsystems like repository, do we have to list them all here?

Advantage is of course that it is easier to handle name clashes, but it always amazes me how easy it is to make names pretty unique without having to resort to UUIDs. France has 60,000 cities and villages, and although there are 25 Beaulieu's (beautiful place), clashes are actually quite rare.

Reply:
Removed the RepoTags enum and added `Constants.REPOTAGS_RESOLVE` and `aQute.bnd.service.RepositoryPlugin.DEFAULT_REPO_TAGS`

Thus, 
- Tags don't know about Repos
- but Repos know about tags